_summaries/he2019momentum_2.md

```diff
@@ -3,17 +3,100 @@ layout: summary
+
+![MoCo Model](./he2019momentum_2a.jpeg)
+
+(queue is not shown in this diagram)
+
+- You have to have sufficient numbers of negative samples to get good representations
+
+- At the same time, you don't want to compare keys that were encoded by an outdated previous iteration of the network 
+
+- A queue allows you to accumulate negative samples, but age out sufficiently old samples that were generated by an older version of the network
+
+- Momentum allows you to use the parameter updates from the query encoder on the key encoder but still produce consistant encodings for the keys.
+
+$$\theta_k \leftarrow m\theta_k + (1-m)\theta_q$$
+
+- where $$\theta_k$$ is the Key Encoder network and $$theta_q$$ is the Query encoding network, $$m$$ is how much momentum the query encoders backprop values are allowed to influence the key encoder.
+
+- MoCo introduces both of these ideas in their implementation.
+
+## How does it perform
+
+- MoCo was trained on Image Net 1k (authors call it 1m because each image is a "class" in their pretext task) as well as Instagram 1 Billion (940million public images)
+
+- MoCo was evaluated agianst other pretraining models on the same pretext task.
+
+- MoCo was evaluated against supervised pretraining and randomly initialized models for downstream tasks 
+    - Because MoCo is fine-tuned on downstream tasks, the experiment was setup, so the fine-tuning was as long as the supervised/random initialized training so that direct comparisons could take place.
+
+
+
+- Extremely well!  It outperforms all other unsupervised pretext models.
+
+<p align="center">
+  <img width="1200" alt="Pretext Tasks" src="./he2019momentum_2d.png">
+</p>
+
+- MoCo also outperforms every model on nearly every down stream task (note the one failure being semantic segmentations on VOC)
+
+<p align="center">
+  <img width="1200" alt="Downstream Tasks" src="./he2019momentum_2c.png">
+</p>
+
+## Does the pretext task or backbone matter (other variants)
+
+- The pretext task does matter, by tweaking the data augmentation the authors were able to get large performance gains (10%+) on the pretext task.
+
+- MoCo is also capable of scaling up to extremely large unsupervised datasets, and performs better consistently as the pretext dataset increases.
+
+- The backbone does matter as well, the reason is unknown at the time of the paper.
+
+<p align="center">
+  <img width="1500" alt="MoCo compared with IG 1B vs IN 1M and C50 vs FPN" src="./he2019momentum_2b.png">
+</p>
+
 
 ## TL;DR
-* Three
-* Bullets
-* To highlight the core concepts
+
+* Contrastive Loss benefits from a large queue-like dictionary of negative samples, where sufficiently old activations are dropped
+
+* Contrastive Loss benefits from momentum weights on the key encoder to produce better keys and thus help the query encoder train better representations
+
+* MoCo and Pretext tasks are now better at pretraining computer vision models than supervised alternatives
```

Comment:
Good summary, you explain things in an easy to understand manner. Left a few comments on things that might be useful to include.

Reply:
Thanks!